src/main/java/teammates/common/datatransfer/questions/FeedbackQuestionDetails.java

```diff
@@ -39,26 +39,36 @@ protected FeedbackQuestionDetails(FeedbackQuestionType questionType, String ques
 
     public abstract String getQuestionTypeDisplayName();
 
+    @Deprecated
     public abstract String getQuestionWithExistingResponseSubmissionFormHtml(
             boolean sessionIsOpen, int qnIdx, int responseIdx, String courseId,
             int totalNumRecipients, FeedbackResponseDetails existingResponseDetails, StudentAttributes student);
 
+    @Deprecated
     public abstract String getQuestionWithoutExistingResponseSubmissionFormHtml(
                                 boolean sessionIsOpen, int qnIdx, int responseIdx, String courseId,
                                 int totalNumRecipients, StudentAttributes student);
 
+    @Deprecated
     public abstract String getQuestionSpecificEditFormHtml(int questionNumber);
 
+    @Deprecated
     public abstract String getNewQuestionSpecificEditFormHtml();
 
+    @Deprecated
     public abstract String getQuestionAdditionalInfoHtml(int questionNumber, String additionalInfoId);
 
+    @Deprecated
     public abstract String getQuestionResultStatisticsHtml(List<FeedbackResponseAttributes> responses,
                                                            FeedbackQuestionAttributes question,
                                                            String studentEmail,
                                                            FeedbackSessionResultsBundle bundle,
                                                            String view);
 
+    public abstract String getQuestionResultStatisticsJson(
```

Comment:
Let's use polymorphism here? Return a class (functioned as DTO) (e.g. `NumericalScaleResultStatistics`) of some base class (e.g. `ResultStatistics`) and let the GSON serialise to different string JSON based on different class?

Reply:
The parameter should not be `studentEmail` as we use `isStudent` flag?

Reply:
> Let's use polymorphism here? Return a class (functioned as DTO) (e.g. `NumericalScaleResultStatistics`) of some base class (e.g. `ResultStatistics`) and let the GSON serialise to different string JSON based on different class?

I don't want to introduce ~10 empty classes yet until I have a clearer idea on the necessary abstract methods.